### PR TITLE
[Components with more than 1 service] LPS-186808 Refactor components from adaptive-media, analytics, layout, marketplace, portal-scheduler, portal-security-sso, portal-template, portal-template and segments modules

### PR DESCRIPTION
**Pull request by RenanUmV**
[LPS-186808](https://issues.liferay.com/browse/LPS-186808)

### CODE REVIEW

File: modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/processor/AMImageProcessorImpl.java [hunk 1]
```diff
@@ -44,7 +43,7 @@
  */
 @Component(
 	property = "model.class.name=com.liferay.portal.kernel.repository.model.FileVersion",
-	service = {AMImageProcessor.class, AMProcessor.class}
+	service = AMImageProcessor.class
 )
```
Comment (rafaprax) [inline modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/processor/AMImageProcessorImpl.java:47]: Hey @RenanUmV, there is a serviceTracker that tracker `AMImageProcessor` here `AMMessageListener`. So, this fix will break the logic there. Please try to find another way the get rid of these multi services.

Thx

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by RenanUmV** [comment]
ci:test:sf

**Comment by RenanUmV** [comment]
ci:test:relevant

File: modules/apps/adaptive-media/adaptive-media-document-library-thumbnails/src/main/java/com/liferay/adaptive/media/document/library/thumbnails/internal/processor/AMImageEntryProcessor.java [hunk 2]
```diff
@@ -64,34 +59,10 @@
...
-		FileVersion sourceFileVersion, FileVersion destinationFileVersion) {
-	}
-
-	@Override
-	public void exportGeneratedFiles(
-		PortletDataContext portletDataContext, FileEntry fileEntry,
-		Element fileEntryElement) {
-	}
+public class AMImageEntryProcessor
+	extends AMDLProcessor implements ImageProcessor {
```
Comment (rafaprax) [inline modules/apps/adaptive-media/adaptive-media-document-library-thumbnails/src/main/java/com/liferay/adaptive/media/document/library/thumbnails/internal/processor/AMImageEntryProcessor.java:65]: Hey @RenanUmV, do not extend the `AMDLProcessor` here... just copy the fields amImageMimeTypeProvider and  amImageValidator here.

Thx

File: modules/apps/adaptive-media/adaptive-media-document-library-thumbnails/src/main/java/com/liferay/adaptive/media/document/library/thumbnails/internal/processor/AMDLProcessor.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	public void trigger(
+		FileVersion sourceFileVersion, FileVersion destinationFileVersion) {
+	}
+
+	protected boolean isMimeTypeSupported(String mimeType) {
+		return amImageMimeTypeProvider.isMimeTypeSupported(mimeType);
+	}
+
+	@Reference
+	protected AMImageMimeTypeProvider amImageMimeTypeProvider;
```
Comment (rafaprax) [inline modules/apps/adaptive-media/adaptive-media-document-library-thumbnails/src/main/java/com/liferay/adaptive/media/document/library/thumbnails/internal/processor/AMDLProcessor.java:89]: make these fields private. Thx

**Review by rafaprax — CHANGES_REQUESTED** [review]

File: modules/apps/adaptive-media/adaptive-media-document-library-thumbnails/src/main/java/com/liferay/adaptive/media/document/library/thumbnails/internal/processor/AMImageEntryProcessor.java (hunk 2 above)
Comment (rafaprax) [reply modules/apps/adaptive-media/adaptive-media-document-library-thumbnails/src/main/java/com/liferay/adaptive/media/document/library/thumbnails/internal/processor/AMImageEntryProcessor.java:65]: Hey @RenanUmV, never mind. As we discussed today, it will require implementing some methods so It's not worth it in my point of view.

Thanks for catch that

**Comment by RenanUmV** [comment]
ci:test:dl-store

**Comment by catarineaguiar** [comment]
ci:test:lima

**Comment by catarineaguiar** [comment]
ci:test:echo

**Comment by catarineaguiar** [comment]
ci:test:security

**Comment by catarineaguiar** [comment]
ci:test:analytics-cloud-acceptance

**Comment by Kvale1733** [comment]
@RenanUmV 
Regressions found

1) com.liferay.adaptive.media.document.library.thumbnails.internal.osgi.commands.test.AMThumbnailsOSGiCommandsTest
```
java.lang.IllegalStateException: Timed out while waiting for service com.liferay.document.library.kernel.util.DLProcessor type=ImageProcessor
		at com.liferay.portal.test.rule.InjectTestBag._getServiceReference(InjectTestBag.java:199)
		at com.liferay.portal.test.rule.InjectTestBag.injectFields(InjectTestBag.java:78)
		at com.liferay.portal.test.rule.InjectTestRule.beforeClass(InjectTestRule.java:49)
		at com.liferay.portal.test.rule.InjectTestRule.beforeClass(InjectTestRule.java:24)
		at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:54)
		at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
```
[Log](https://testray.liferay.com/reports/production/logs/2023-06/test-1-40/test-portal-acceptance-pullrequest(master)/3095/modules-integration-mysql57-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

2) com.liferay.adaptive.media.document.library.thumbnails.internal.processor.AMImageEntryProcessorTest
```
 [exec] com.liferay.adaptive.media.document.library.thumbnails.internal.processor.AMImageEntryProcessorTest > testGetPreviewFileSizeDoesNotTriggerAMProcessorWhenAMImageExists FAILED
     [exec]     java.lang.NoSuchFieldException: No field on null with name _amImageMimeTypeProvider
     [exec]         at com.liferay.portal.kernel.test.ReflectionTestUtil.getField(ReflectionTestUtil.java:135)
     [exec]         at com.liferay.portal.kernel.test.ReflectionTestUtil.setFieldValue(ReflectionTestUtil.java:346)
     [exec]         at com.liferay.adaptive.media.document.library.thumbnails.internal.processor.AMImageEntryProcessorTest.setUp(AMImageEntryProcessorTest.java:69)
     [exec]         at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     [exec]         at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
     [exec]         at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
     [exec]         at java.lang.reflect.Method.invoke(Method.java:498)
     [exec]         at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
     [exec]         at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
```
[Log](https://testray.liferay.com/reports/production/logs/2023-06/test-1-40/test-portal-acceptance-pullrequest(master)/3095/modules-unit-jdk8/0/0/jenkins-console.txt.gz?authuser=0)


3) Error on shutdown. Just startup and shutdown. The following should appear in a log file in osgi/state
```
!ENTRY com.liferay.portal.search 4 0 2023-06-14 21:03:50.938
!MESSAGE bundle com.liferay.portal.search:8.0.73 (263)[com.liferay.portal.search.internal.background.task.BackgroundTaskExecutorConfigurator(549)] : The deactivate method has thrown an exception
!STACK 0
java.lang.IllegalStateException: The service has been unregistered
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.unregister(ServiceRegistrationImpl.java:204)
	at com.liferay.portal.search.internal.background.task.BackgroundTaskExecutorConfigurator.deactivate(BackgroundTaskExecutorConfigurator.java:58)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflec
```

4) Images not shown on image fragment


https://github.com/liferay-core-infra/liferay-portal/assets/90647688/d1f73fca-a8d7-4ffd-870c-35fff04dc8c0

5) Regression in saml test.SAML#AssertSSOWithEncryptionAndDecryption. 

Seems this tests is supposed to return "Data encryption key cannot be null" but now it returns "Encryption is forced for samlsp, but no encryption parameters have been successfully negotiated" [Log](https://testray.liferay.com/reports/production/logs/2023-06/test-1-34/test-portal-acceptance-pullrequest(master)/4230/functional-tomcat90-mysql57-jdk8/0/14/jenkins-console.txt.gz?authuser=0).

Here's a recording of the test being run to show the steps. The only thing not shown since it's done with JSON is to add virtual instance www.able.com first.

https://drive.google.com/file/d/1-wCDy3gAU_AW0EbGRn0-I8ZQUSRVORwR/view?usp=sharing

**Comment by RenanUmV** [comment]
ci:test:sf

**Comment by RenanUmV** [comment]
ci:test:sf

**Comment by RenanUmV** [comment]
ci:test:relevant

**Comment by RenanUmV** [comment]
ci:test:dl-store

**Comment by RenanUmV** [comment]
ci:test:echo

**Comment by RenanUmV** [comment]
ci:test:security

**Comment by RenanUmV** [comment]
ci:test:analytics-cloud-acceptance

**Comment by RenanUmV** [comment]
ci:test:sf

**Comment by RenanUmV** [comment]
ci:test:relevant

**Comment by RenanUmV** [comment]
ci:test:analytics-cloud-acceptance

**Comment by RenanUmV** [comment]
liferay/ci:test:dl-store

**Comment by RenanUmV** [comment]
liferay/ci:test:echo

**Comment by RenanUmV** [comment]
ci:test:dl-store

**Comment by RenanUmV** [comment]
ci:test:echo

**Comment by RenanUmV** [comment]
ci:test:lima

File: modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/background/task/BackgroundTaskExecutorConfigurator.java [hunk 4]
```diff
@@ -46,9 +46,6 @@ protected void activate(BundleContext bundleContext) {
 
 		_registerBackgroundTaskExecutor(
 			bundleContext, reindexPortalBackgroundTaskExecutor);
-
```
Comment (rafaprax) [inline modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/background/task/BackgroundTaskExecutorConfigurator.java:49]: Please put this removal change in another commit and explain the reasons for that

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by catarineaguiar** [comment]
ci:test:relevant
